todo_app/app.py

```diff
@@ -1,29 +1,88 @@
 from flask import Flask, redirect, render_template, request
+import requests
 from todo_app.data.session_items import add_item, get_item, get_items, save_item
-
+import os
 from todo_app.flask_config import Config
+from todo_app.models.task import Task
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
-
 @app.route('/')
 def index():
-    all_items = get_items()
-    todos, completed = [], []
-    for item in all_items:
-        (completed if item['status'] == 'Completed' else todos).append(item)
-    return render_template("index.html", todos=todos, completed=completed)
+    board_id = os.getenv('BOARD_ID')
+    cards_path = f"/boards/{board_id}/cards"
+    cards = requests.get(full_url(cards_path)).json()
+    tasks = [Task.from_trello_card(card) for card in cards]
+    todos = [task for task in tasks if task.not_started()]
+    in_progress = [task for task in tasks if task.in_progress()]
+    complete = [task for task in tasks if task.complete()]
+    tasks_without_status = [task for task in tasks if task.statusless()]
+    tasks_with_multiple_stati = [task for task in tasks if task.has_multiple_stati()]
+
+    return render_template(
+        "index.html",
+        todos=todos,
+        in_progress=in_progress,
+        complete=complete,
+        tasks_without_status=tasks_without_status,
+        tasks_with_multiple_stati=tasks_with_multiple_stati
+    )
 
 @app.route('/add-todo', methods=['POST'])
 def create():
     todo_title = request.form.get('title')
-    add_item(todo_title)
+    new_card_path = "/cards"
+    query = {
+        "name": todo_title,
+        "idList": os.getenv("LIST_ID"),
+    }
+    requests.request(
+        "POST",
+        full_url(new_card_path),
+        headers={"Accept": "application/json"},
+        params=query,
+    )
+    return redirect('/')
+
+@app.route('/change-status/<status>/<task_id>', methods=["POST"])
+def change_status(task_id, status):
+    status_ids = get_status_label_ids()
+    status_id = status_ids[status]
+    add_status(task_id, status_id)
+    return redirect('/')
+
+def full_url(path):
+    trello_api_root = "https://api.trello.com/1"
+    trello_api_key = os.getenv('TRELLO_API_KEY')
+    trello_api_token = os.getenv('TRELLO_API_TOKEN')
+    return f"{trello_api_root}{path}?key={trello_api_key}&token={trello_api_token}"
+
+def find_status(trello_card):
+    labels = trello_card["labels"]
+    status_options = ['Not Started', 'In Progress', 'Complete']
+    statuses = [label["name"] for label in labels if label["name"] in status_options]
+    status_count = len(statuses)
+    if status_count == 1:
+        return True, statuses[0]
+    elif status_count < 1:
+        return False, None
+    else:
+        return False, statuses
```

Comment:
I think this function hasn't been used in the final version, can we delete it?
```suggestion
```

Reply:
vg. good spot